core/src/impl/Kokkos_ViewMapping.hpp

```diff
@@ -3430,8 +3386,15 @@ class ViewMapping<
     // Query the mapping for byte-size of allocation.
     // If padding is allowed then pass in sizeof value type
     // for padding computation.
-    using padding = std::integral_constant<
-        unsigned int, alloc_prop::allow_padding ? sizeof(value_type) : 0>;
+
+    using use_padding =
+        std::bool_constant<alloc_prop::allow_padding &&
+                           !std::is_same<typename Traits::array_layout,
+                                         Kokkos::LayoutStride>::value>;
```

Comment:
```suggestion
                           !std::is_same_v<typename Traits::array_layout,
                                         Kokkos::LayoutStride>>;
```

Reply:
We should update this across files (Kokkos_View, Kokkos_ViewMapping, etc.) so I will defer this to another PR.